iip-5.md

```diff
@@ -0,0 +1,52 @@
+```
+IIP: 5
+Title: Classifying EVM Error Status
+Author: Dorothy Ko (dorothy@iotex.io)
+Status: Draft
+Type: Standards Track
+Created: 2019-08-14
+```
+
+
+## Abstract
+Currently every execution of smart contract on IoTex Blockchain doesn't show any information of error if it fails to execute or deploy the smart contract. In order to give an information of error to the end-user, this standard outlines a common set of EVM Error Status Code. Therefore, we classify the possible EVM error and define EVM error status code. And we propose to give the EVM Error Code to the Receipt Status, so that it is visible to users/developers and accordingly it enhances UX(User Experience) and DX(Developer Experience). 
+
+## Motivation
+Sometimes, we failed to execute or deploy a smart contract. However, currently its status gives us just simple status such as Success(1) or Failure(0) in receipt and it doesn't give us any information of the reason of failure. In terms of user and developer experience, it makes us difficult to utilize the smart contract execution.  
+
+## Specification
+
+> General Status 
+
+| code | Description  |
+|------|--------------|
+| 0    | Failure      |
+| 1    | Success      |
+
+
+> 1xx for EVM Execution Error Status 
+
+| code | 						Description                               |
+|------|------------------------------------------------------------------|
+| 100  | ErrUnknown : Unknown                                             |
+| 101  | ErrOutOfGas : Out of Gas                                         |
+| 102  | ErrCodeStoreOutOfGas : Contract creation code storage out of gas |
+| 103  | ErrDepth : Max call depth exceeded                               |
+| 104  | ErrContractAddressCollision : Contract address collision         |
```

Comment:
this should happen with a negligible chance. did we see this often?

Reply:
I don't think it often happens, but this 1xx error code level is evm-level which is lower than EIP1066, so I think it may need all possible error status.. Most frequent error is outofgas or reverted I think